File: data/src/main/java/com/data/net/repository/datasource/NetDataSource.java
package com.data.net.repository.datasource;

import com.data.net.retrofit.RetrofitApiCreator;
import com.data.net.retrofit.RetrofitService;
import com.data.net.sharedprefs.PrefsDataSource;

public class NetDataSource {

    private final RetrofitService mRetrofitService;

    public NetDataSource(PrefsDataSource sharedPreferences,
                         RetrofitApiCreator retrofitApiCreator) {
        mRetrofitService = retrofitApiCreator.create(sharedPreferences);
    }

//    public Flowable<BaseResponse<String>> getPrivacyPolicyUrl() {
//        return mRetrofitClientService.getPrivacyPolicyUrl();
//    }

}

File: data/src/main/java/com/data/net/repository/datasource/NetDataSource.kt
package com.data.net.repository.datasource

import com.data.net.pojo.User
import com.data.net.retrofit.BaseResponse
import com.data.net.retrofit.RetrofitApiCreator
import io.reactivex.Flowable

open class NetDataSource(prefsDataSource: PrefsDataSource, retrofitApiCreator: RetrofitApiCreator) {
    private val mRetrofitService = retrofitApiCreator.create(prefsDataSource)

    fun register(map: Map<String, String>): Flowable<BaseResponse<User?>> {
        return mRetrofitService.register(map)
    }

}
